sdk/cognitiveservices/cognitiveservices-textanalytics/review/cognitiveservices-textanalytics.api.md

```diff
@@ -0,0 +1,346 @@
+    length: number;
+    offset: number;
+    sentenceScores: any;
+    sentiment: Sentiment;
+    warnings?: string[];
+}
+
+// @public
+export type Sentiment = 'positive' | 'neutral' | 'negative';
+
+// @public (undocumented)
+export interface TextAnalysisErrorResult {
+    readonly error: ErrorModel;
+    readonly id: string;
+}
+
+// @public (undocumented)
+export type TextAnalysisResult = TextAnalysisSuccessResult | TextAnalysisErrorResult;
+
+// @public (undocumented)
+export interface TextAnalysisSuccessResult {
+    readonly id: string;
+    readonly statistics?: DocumentStatistics;
+}
+
+// @public
+export class TextAnalyticsClient {
+    constructor(endpointUrl: string, credential: TokenCredential | CognitiveServicesCredentials, options?: TextAnalyticsClientOptions);
+    // (undocumented)
+    analyzeSentiment(input: string[], language?: string, options?: AnalyzeSentimentOptions): Promise<AnalyzeSentimentResultCollection>;
+    // (undocumented)
+    analyzeSentiment(input: MultiLanguageInput[], options?: AnalyzeSentimentOptions): Promise<AnalyzeSentimentResultCollection>;
+    defaultCountryHint: string;
+    defaultLanguage: string;
+    // (undocumented)
+    detectLanguage(input: string[], countryHint?: string, options?: DetectLanguagesOptions): Promise<DetectLanguageResultCollection>;
+    // (undocumented)
+    detectLanguage(input: LanguageInput[], options?: DetectLanguagesOptions): Promise<DetectLanguageResultCollection>;
+    readonly endpointUrl: string;
+    // (undocumented)
+    extractEntityLinking(input: string[], language?: string, options?: ExtractEntityLinkingOptions): Promise<ExtractLinkedEntitiesResultCollection>;
+    // (undocumented)
+    extractEntityLinking(input: MultiLanguageInput[], options?: ExtractEntityLinkingOptions): Promise<ExtractLinkedEntitiesResultCollection>;
+    // (undocumented)
+    extractKeyPhrases(input: string[], language?: string, options?: ExtractKeyPhrasesOptions): Promise<ExtractKeyPhrasesResultCollection>;
+    // (undocumented)
+    extractKeyPhrases(input: MultiLanguageInput[], options?: ExtractKeyPhrasesOptions): Promise<ExtractKeyPhrasesResultCollection>;
+    // (undocumented)
+    recognizeEntities(input: string[], language?: string, options?: RecognizeEntitiesOptions): Promise<RecognizeEntitiesResultCollection>;
+    // (undocumented)
+    recognizeEntities(input: MultiLanguageInput[], options?: RecognizeEntitiesOptions): Promise<RecognizeEntitiesResultCollection>;
+    // (undocumented)
+    recognizePiiEntities(input: string[], language?: string, options?: RecognizePiiEntitiesOptions): Promise<RecognizeEntitiesResultCollection>;
+    // (undocumented)
+    recognizePiiEntities(input: MultiLanguageInput[], options?: RecognizePiiEntitiesOptions): Promise<RecognizeEntitiesResultCollection>;
+    // (undocumented)
+    singleAnalyzeSentiment(inputText: string, language?: string, options?: AnalyzeSentimentOptions): Promise<AnalyzeSentimentSuccessResult>;
```

Comment:
I don't like the `single*` approach. I think I'd rather either overloads with different throw behavior, or even overloads where singular inputs return singular outputs (e.g. a RecognizeEntitiesResult).

Reply:
will ponder this weekend and get back to you